hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/processor/Processors.java

```diff
@@ -767,6 +771,26 @@ public static ProcessorSupplier nonCooperativeP(@Nonnull ProcessorSupplier wrapp
         };
     }
 
+    /**
+     * Returns a supplier of processors for a vertex that implements
+     * hash join transform.
+     *
+     * @param keyFns           list of functions which will be used to extract a separate key
+     *                         for each of the lookup tables
+     * @param tags             used to populate the output items
+     * @param mapToOutputBiFn  function which will compute the emitting item (2-way)
+     * @param mapToOutputTriFn function which will compute the emitting item (3-way)
+     * @param <E0>             type of the received item
+     */
+    public static <E0> DistributedSupplier<Processor> hashJoinP(
```

Comment:
Either remove this from public API or it needs lot more documentation and also a publich method for `HashJoinCollectP`.